x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/InlineStats.java

```diff
@@ -151,6 +163,30 @@ public LogicalPlan firstPhase() {
 
     @Override
     public LogicalPlan nextPhase(List<Attribute> schema, List<Page> firstPhaseResult) {
+        if (groupings.isEmpty()) {
+            return ungroupedNextPhase(schema, firstPhaseResult);
+        }
+        return groupedNextPhase(schema, firstPhaseResult);
+    }
+
+    private LogicalPlan ungroupedNextPhase(List<Attribute> schema, List<Page> firstPhaseResult) {
+        if (firstPhaseResult.size() != 1) {
+            throw new IllegalArgumentException("expected single row");
+        }
+        Page p = firstPhaseResult.get(0);
+        if (p.getPositionCount() != 1) {
+            throw new IllegalArgumentException("expected single row");
+        }
+        List<Alias> values = new ArrayList<>(schema.size());
+        for (int i = 0; i < schema.size(); i++) {
+            Attribute s = schema.get(i);
+            Object value = BlockUtils.toJavaObject(p.getBlock(i), 0);
+            values.add(new Alias(source(), s.name(), null, new Literal(source(), value, s.dataType()), aggregates.get(i).id()));
+        }
+        return new Eval(source(), child(), values);
```

Comment:
Yeahp, this should work and is conceptually a bit nicer (IMHO) than doing that down in the physical mapping!